spikewrap/data_classes/preprocessing.py

```diff
@@ -0,0 +1,58 @@
+import shutil
+from dataclasses import dataclass
+from typing import Dict
+
+import spikeinterface
+
+from ..utils import utils
+from .base import BaseUserDict
+
+
+@dataclass
+class PreprocessingData(BaseUserDict):
+    """
+    Dictionary to store SpikeInterface preprocessing recordings.
+
+    Details on the preprocessing steps are held in the dictionary keys e.g.
+    e.g. 0-raw, 1-raw-bandpass_filter, 2-raw_bandpass_filter-common_average
+    and recording objects are held in the value. These are generated
+    by the `pipeline.preprocess.run_preprocessing()` function.
```

Comment:
I guess the method `pipeline.preprocess.run_preprocessing()` is not part of this PR

Reply:
This will be next PR 😅